tests/deseq2/test_dispersions.py

```diff
@@ -0,0 +1,155 @@
+                                            modelMatrix=x)["betaMatrix"]
+        log_alpha_prior_mean = .5
+        log_alpha_prior_sigmasq = 1
+        mu_hat = np.exp(x @ betaDESeq.T).values.squeeze()
+
+        dispRes = fitDisp(y[:,None],
+                          x=x,
+                          mu_hat=mu_hat[:,None],
+                          log_alpha=0,
+                          log_alpha_prior_mean=log_alpha_prior_mean,
+                          log_alpha_prior_sigmasq=log_alpha_prior_sigmasq,
+                          min_log_alpha=np.log(1e-8),
+                          kappa_0=1,
+                          tol=1e-16,
+                          maxit=100,
+                          usePrior=True,
+                          weights=np.ones((len(y), 1)),
+                          useWeights=False,
+                          weightThreshold=1e-2,
+                          useCR=True)
+
+        # maximum a posteriori (MAP) estimate from DESeq
+        dispDESeq = dispRes["log_alpha"]
+
+        # MAP estimate using optim
+        def logPost(log_alpha):
+            alpha = np.exp(log_alpha)
+            w = np.diag(1 / (1 / mu_hat**2 * (mu_hat + alpha * mu_hat**2)))
+            logLike = np.sum(dnbinom_mu(y, mu=mu_hat, size=1/alpha, log=True))
+            coxReid = -.5*np.linalg.slogdet(x.T @ w @ x)[1]
+            logPrior = dnorm(log_alpha, log_alpha_prior_mean, np.sqrt(log_alpha_prior_sigmasq), log=True)
+            res = logLike + coxReid + logPrior
+            return res
+
+        dispOptim = minimize(lambda p: -logPost(p), 0).x
+
+        self.assertTrue(np.allclose(dispDESeq, dispOptim))
+
+        # check derivatives:
+
+        # from Ted Harding https://stat.ethz.ch/pipermail/r-help/2007-September/140013.html
+        def num_deriv(f, x, h=0.001):
+            return (f(x + h/2) - f(x - h/2)) / h
+        def num_2nd_deriv(f, x, h=0.001):
+            return (f(x + h) - 2*f(x) + f(x - h)) / h**2
+
+        # first derivative of log posterior wrt log alpha at start
+        dispDerivDESeq = dispRes["initial_dlp"]
+        dispDerivNum = num_deriv(logPost, 0)
+        self.assertTrue(np.allclose(dispDerivDESeq, dispDerivNum))
+
+        # second derivative at finish
+        dispD2DESeq = dispRes["last_d2lp"]
+        dispD2Num = num_2nd_deriv(logPost, dispRes["log_alpha"])
+        self.assertTrue(np.allclose(dispD2DESeq, dispD2Num))
+
+        # test fit alternative
+        dds = makeExampleDESeqDataSet()
+        dds = dds.estimateSizeFactors()
+        #ddsLocal = dds.copy().estimateDispersions(fitType="local")
+        ddsMean = dds.copy().estimateDispersions(fitType="mean")
+        ddsMed = estimateDispersionsGeneEst(dds.copy())
+        useForMedian = ddsMed.var["dispGeneEst"] > 1e-7
+        medianDisp = np.nanmedian(ddsMed.var["dispGeneEst"][useForMedian])
+        ddsMed.setDispFunction(lambda x: medianDisp)
+        ddsMed = estimateDispersionsMAP(ddsMed)
```

Comment:
Why is there not assertion on this object if we want to try it?

Reply:
The intent here is rather to test that no error is raised, that `estimateDispersionsMAP` has enough info to run